src/ui/component/splash/view.jsx

```diff
@@ -118,27 +118,29 @@ export default class SplashScreen extends React.PureComponent<Props, State> {
     const { wallet, startup_status: startupStatus, blockchain_headers: blockchainHeaders } = status;
 
     // If the wallet is locked, stop doing anything and make the user input their password
-    if (status.is_running && wallet && wallet.is_locked) {
+    if (status.is_running) {
       // Clear the error timeout, it might sit on this step for a while until someone enters their password
       if (this.timeout) {
         clearTimeout(this.timeout);
       }
 
-      // Make sure there isn't another active modal (like INCOMPATIBLE_DAEMON)
-      if (launchedModal === false && !modal) {
-        this.setState({ launchedModal: true }, () => notifyUnlockWallet());
-      }
-    } else if (status.is_running) {
-      // If we cleared the error timeout due to a wallet being locked, make sure to start it back up
-      if (!this.timeout) {
-        this.adjustErrorTimeout();
-      }
-
-      Lbry.resolve({ urls: 'lbry://one' }).then(() => {
-        this.setState({ isRunning: true }, () => this.continueAppLaunch());
+      Lbry.wallet_status().then(walletStatus => {
+        if (walletStatus.is_locked) {
+          // Make sure there isn't another active modal (like INCOMPATIBLE_DAEMON)
```

Comment:
this is a fine solution but I think ideally this would instead be in the style of `main--auth-page` rather than a modal